### ISSUE TITLE
13x13 and 9x9 nets

### ISSUE THREAD

**Issue by ihavnoid** [ihavnoid/leelaz-ninenine#6]
9x9 - https://drive.google.com/open?id=19QnzpENWRWk-c97HyV4NC2P-VEYE9OGd
13x13 - https://drive.google.com/open?id=1knpQd7wSC8Slqnz_Jro9VnhqQMT5sovE

These are the nets I created using three GTX 1080s - 9x9 took roughly 3 weeks, 13x13 took roughly 8 weeks and still running.

**Comment by alreadydone** [comment]
Your nets are 128x10 and 192x15 respectively. These are trained with 7.5 komi, right?
@hzyhhzy tested (with --noponder -p50 -r10 -t1 -m10) the 9x9 net with his series of nets and found that it's roughly on par with (slightly stronger than) his 20th net (64x5, trained with -v 50 on 1x1080Ti for 70 hours).
Here is the 48th net (96x5, 2 weeks) for comparison. 
https://drive.google.com/file/d/1s9WLzt2lfuWbJfS_0TAyj08A6wX7CMpU/view?usp=sharing
He's now at 149th net (224x16) with -p 50. 
https://drive.google.com/file/d/1ViWarBvxhzMdvyC8IACQ-jahL03LBD7m/view?usp=sharing
He advises using a smaller window at the beginning to speed things up and increasing noise significantly to allow the opening moves to change.

**Comment by ihavnoid** [comment]
Yeah 7.5 komi.  I haven't paid a lot of attention on the parameters so it is possible that it ended up being stuck somewhere suboptimal.

To me it's a bit surprising that 50 playoffs are enough.  That definitely will speed up things by a lot.  I used 400.

**Comment by ihavnoid** [comment]
FYI, the 13x13 was trained using 800 playouts.  I didn't pay a lot of attention to the parameters, so again, what I did may not be anything close to optimal.  :)  It at least beats me 10-0 very easily.

**Comment by roy7** [comment]
Is there a Leela Zero for Windows binary someplace that can be used for 13x13?

**Comment by roy7** [comment]
Thanks, someone on Discord provided one to the person on Discord who was asking (I posted here to ask for him). :)

**Comment by trunterzx** [comment]
@roy7 

Is it possible to share the link? Mine doesn't seem to be working very well due to some reasons...

> Thanks, someone on Discord provided one to the person on Discord who was asking (I posted here to ask for him). :)

**Comment by alreadydone** [comment]
[lz13.zip](https://github.com/ihavnoid/leelaz-ninenine/files/2799790/lz13.zip)

**Comment by roy7** [comment]
The one shared on Discord:

```I compiled a lz exe for 13×13, but it might be a little bit outdated. You need to download the leelaz 0.15 and replace the leelaz.exe with this one.
https://share.weiyun.com/516RHoY
And here is a weight for 13×13. I have tested it with a pro player and it won 2-0.
https://share.weiyun.com/5f8mdIA```

**Comment by hzyhhzy** [comment]
My latest 9x9 weight(komi 7.5,30b320f).16b224f seems to 
https://drive.google.com/open?id=1I7LtMcuwyLUZDNjV5VL9eUwTDdyA2nIT

**Comment by trunterzx** [comment]
@alreadydone @roy7 

Thanks for sharing!

Is this compatible with Lizzie0.6? I tried replacing the leelaz.exe in Lizzie0.6 with this one, but it didn't seem to work... Not sure whether it's just my computer...

**Comment by roy7** [comment]
I assume you'd need to recompile Lizzie for alternate board sizes. You may need to use Sabaki for a GUI.

**Comment by trunterzx** [comment]
@hzyhhzy 

Thanks for sharing!

Maybe 7.5 is too much komi for black on 9x9... Perhaps 6.5 or 5.5 will be a fairer komi?


> My latest 9x9 weight(komi 7.5,30b320f).
> Black's winrate is about 25% in self-play games now.Maybe white will never lose someday.

**Comment by hzyhhzy** [comment]
> @hzyhhzy
> 
> Thanks for sharing!
> 
> Maybe 7.5 is too much komi for black on 9x9... Perhaps 6.5 or 5.5 will be a fairer komi?
> 
> > My latest 9x9 weight(komi 7.5,30b320f).
> > Black's winrate is about 25% in self-play games now.Maybe white will never lose someday.

No,the fairest komi must be an integer(most likely an odd number),and it is probably 7.0 according to research.

**Comment by kennychenfs** [comment]
No, I don't think so.
The komi was designed for a decimal because it won't make draw game.

**Comment by alreadydone** [comment]
Note that major computer Go competitions as well as CGOS uses 7.0 komi for 9x9 games, so it must be empirically clear that the fair komi is 7.0. Of course there's still a possibility that not losing the game is easier for one color than for the other. Even if a non-losing strategy is found, it may take more space to store / more time to compute for one color than the other.

**Comment by mccxj** [comment]
@ihavnoid  Where to download the latest weight?

**Comment by l1t1** [comment]
good

**Comment by trunterzx** [comment]
Thanks for sharing!


https://github.com/leela-zero/leela-zero/issues/2240#issue-412504088

Is the one shared above stronger than the one shared below?

https://github.com/ihavnoid/leelaz-ninenine/issues/6#issuecomment-457891216


How about the 13x13, is it stronger than the one below?

https://github.com/ihavnoid/leelaz-ninenine/issues/6#issuecomment-457890152

**Comment by hzyhhzy** [comment]
> Thanks for sharing!
> 
> [leela-zero/leela-zero#2240 (comment)](https://github.com/leela-zero/leela-zero/issues/2240#issue-412504088)
> 
> Is the one shared above stronger than the one shared below?
> 
> [#6 (comment)](https://github.com/ihavnoid/leelaz-ninenine/issues/6#issuecomment-457891216)
> 
> How about the 13x13, is it stronger than the one below?
> 
> [#6 (comment)](https://github.com/ihavnoid/leelaz-ninenine/issues/6#issuecomment-457890152)

13x13 modified from 205 is much stronger than that one.But 9x9 is much weaker.

**Comment by l1t1** [comment]
did you try to modified from elfv2? @hzyhhzy

**Comment by hzyhhzy** [comment]
Do it yourself

**Comment by iopq** [comment]
@hzyhhzy 149s is competitive with 152s at time parity

Maybe 30b is overkill for 9x9?

**Comment by alreadydone** [comment]
latest is 158s I think

**Comment by iopq** [comment]
Where do you get 158s?

**Comment by mihhailnovik** [comment]
Tested 152s on GoQuest server against real people. Result - 150+ wins, few draws, 0 lose. 

If anyone interested.. this is a bridge i had to implement to connect GTP and GoQuest https://github.com/mihhailnovik/questGo

**Comment by iopq** [comment]
@hzyhhzy any newer nets? Or should I just post your 169 net here as the final one?

**Comment by hzyhhzy** [comment]
My most recent one is 171. But I think it's meaningless to release it.
Recently katago released 20b weights. I tested and found that the last one is almost as strong as my "171s"(about 200 games, 10000po, komi 7.5 , the winrate of katago is 43%), and it also supports other boardsizes and komis. So I stopped my LZ9 and started to train Katago9.
I had several runs of Katago 9x9 on my computer. It's about 30x faster than my LZ9. It only took me about 3 days on 2*2060 to reach the level of the last Katago 20b on 9x9 board. 
I'm trying some parameters, several weeks later I may release a stronger Katago 9x9 weight.

**Comment by iopq** [comment]
I really need the Katago 9x9 weight, since I play on 7 komi server. Game reviews are not valid with the wrong komi

**Comment by leeazoo** [comment]
Can someone share the network.gz for lizzie to play 13x13, i found this topic, i try with the files you put here, but i cant unrar the rar, and the 13x13.txt.gz , gets engine loading, and lz13 in zip is just for windows,  Can any one share a network trained for 13x13, 

Also, can someone point a tutorial on how to train networks, pls dont be hard with me im just learning.

**Comment by iopq** [comment]
You need to compile the 13x13 Leela Zero yourself by changing the board size from 19 to 13, then it will be able to load the network file

**Comment by leeazoo** [comment]
That was the issue, thanks for the tip.

**Comment by dustinlacewell** [comment]
Does anyone have an updated link to 13x13 weights? It seems the links died.

**Comment by iopq** [comment]
@dustinlacewell use KataGo, the latest nets are stronger
